customer/src/test/java/no/unit/nva/customer/update/UpdateCustomerDoiHandlerTest.java

```diff
@@ -0,0 +1,120 @@
+import java.net.HttpURLConnection;
+import java.util.Map;
+import java.util.UUID;
+import no.unit.nva.customer.model.CustomerDto;
+import no.unit.nva.customer.model.CustomerDto.DoiAgentDto;
+import no.unit.nva.customer.service.CustomerService;
+import no.unit.nva.customer.testing.CustomerDataGenerator;
+import no.unit.nva.stubs.FakeContext;
+import no.unit.nva.testutils.HandlerRequestBuilder;
+import nva.commons.apigateway.GatewayResponse;
+import nva.commons.apigateway.exceptions.ApiGatewayException;
+import nva.commons.apigateway.exceptions.NotFoundException;
+import nva.commons.secrets.SecretsWriter;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.zalando.problem.Problem;
+
+class UpdateCustomerDoiHandlerTest {
+
+    private static final Context CONTEXT = new FakeContext();
+
+    private CustomerDto existingCustomer;
+    private ByteArrayOutputStream outputStream;
+    private CustomerService customerServiceMock;
+    private SecretsWriter secretWriterMock;
+    private UpdateCustomerDoiHandler handler;
+
+    @BeforeEach
+    public void beforeEach() {
+        this.outputStream = new ByteArrayOutputStream();
+        customerServiceMock = mock(CustomerService.class);
+        secretWriterMock = mock(SecretsWriter.class);
+        handler = new UpdateCustomerDoiHandler(customerServiceMock, secretWriterMock);
+        existingCustomer = CustomerDataGenerator.createSampleCustomerDao().toCustomerDto();
+    }
+
+    @Test
+    void handleUpdateRequestReturnsNotFoundWhenARequestWithANonExistingIdentifier()
+        throws ApiGatewayException, IOException {
+        var secret = randomString();
+        var doiAgent = existingCustomer.getDoiAgent()
+                           .addSecret(secret);
+
+        when(customerServiceMock.getCustomer(any(UUID.class)))
+            .thenReturn(existingCustomer);
+
+        when(customerServiceMock.updateCustomer(any(UUID.class), any(CustomerDto.class))
+        ).thenReturn(existingCustomer);
+
+        when(secretWriterMock.updateSecretKey(any(), eq(existingCustomer.getDoiAgent().getPrefix()), any())
+        ).thenReturn(existingCustomer.getDoiAgent().getName());
+
+        var response = sendRequest(getExistingCustomerIdentifier(), doiAgent, DoiAgentDto.class);
+        var doiAgentResponse = response.getBodyObject(DoiAgentDto.class);
+
+        assertThat(response.getStatusCode(), is(equalTo(HttpURLConnection.HTTP_OK)));
+        assertThat(doiAgentResponse.getSecret(), is(equalTo(secret)));
+    }
+
+    @Test
+    void failUpdateRequestReturnsNotFoundWhenARequestWithAInvalidIdentifier()
+        throws ApiGatewayException, IOException {
+        when(
+            customerServiceMock.getCustomer(any(UUID.class))
+        ).thenThrow(NotFoundException.class);
+
+        var response = sendRequest(createRequestWithInvalidUuid(null), Problem.class);
+        assertThat(response.getStatusCode(), is(equalTo(HttpURLConnection.HTTP_BAD_REQUEST)));
```

Comment:
Not Found is bad request? In the other handler's test not found was Bad Gateway?